### PR TITLE
[CodeComplete] Markup default arguments

### PR DESCRIPTION
**Pull request by bnbarham**
Add `<callarg.default />` nodes to the annotated description of a
completion result for all arguments that have defaults.

Also add a new option to control whether or not an extra item is added
for calls with no default arguments or not (since clients can now use
the marked up argument to check this themselves).

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
Windows failure is known (and that test since disabled).

File: lib/IDE/CodeCompletion.cpp [hunk 1]
```diff
@@ -1004,6 +1007,9 @@ void CodeCompletionResultBuilder::addCallArgument(
     addChunkWithText(ChunkKind::CallArgumentType, TypeName);
   }
 
+  if (HasDefault)
+    addSimpleChunk(ChunkKind::CallArgumentDefault);
```
Comment (rintaro) [inline lib/IDE/CodeCompletion.cpp:1011]: If we want to add the actual argument value to the type, we probably want it like `<callarg.default>=  nil</callarg.default>` Does it make sense to make this like
```cpp
withNestedGroup(ChunkKind::CallArgumentDefaultBegin, []() {
  // TODO?: Add the actual value?
})
```

Comment (bnbarham) [reply lib/IDE/CodeCompletion.cpp:1011]: Sure, seems reasonable.

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please test macOS platform

**Comment by bnbarham** [comment]
@swift-ci please clean test macOS platform

File: lib/IDE/CodeCompletionResultPrinter.cpp [hunk 2]
```diff
@@ -187,6 +190,10 @@ class AnnotatingResultPrinter {
         printNestedGroup("callarg.type", i, e);
         continue;
       }
+      if (i->is(ChunkKind::CallArgumentDefaultBegin)) {
+        printNestedGroup("callarg.default", i, e);
+        continue;
+      }
```
Comment (rintaro) [inline lib/IDE/CodeCompletionResultPrinter.cpp:196]: void printNestedGroup(StringRef tag, ChunkIter &i, const ChunkIter e) {
    assert(i != e && !i->hasText() &&
           CodeCompletionString::Chunk::chunkStartsNestedGroup(i->getKind()));
    auto nestingLevel = i->getNestingLevel();
    // Skip the "Begin" chunk.
    ++i;
    if (

    if (!tag.empty())
      OS << "<" << tag << ">";
    while (i != e && !i->endsPreviousNestedGroup(nestingLevel)) {

Comment (bnbarham) [reply lib/IDE/CodeCompletionResultPrinter.cpp:196]: Sure

**Comment by bnbarham** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
@swift-ci please clean test macOS platform

File: lib/IDE/CodeCompletionResultPrinter.cpp [hunk 3]
```diff
@@ -120,6 +120,9 @@ class AnnotatingResultPrinter {
     OS << content.substr(rtrimIdx);
   }
 
+  /// Print an empty \p tag.
+  void printEmptyTag(StringRef tag) { OS << "<" << tag << " />"; }
+
```
Comment (rintaro) [inline lib/IDE/CodeCompletionResultPrinter.cpp:125]: This is unused.

**Review by rintaro — APPROVED** [review]
LGTM! Thank you!

File: lib/IDE/CodeCompletionResultPrinter.cpp (hunk 3 above)
Comment (bnbarham) [reply lib/IDE/CodeCompletionResultPrinter.cpp:125]: Thanks! I'll let tests pass and then remove + smoke check.

**Comment by bnbarham** [comment]
@swift-ci please smoke test

**Pull request merged**